src/main/java/com/imsweb/mph/MphUtils.java

```diff
@@ -201,17 +186,22 @@ public static MphGroup findCancerGroup(String primarySite, String histology, Str
     /**
      * @return the list of cancer groups.
      */
-    public static List<MphGroup> getAllGroups() {
-        return Collections.unmodifiableList(_GROUPS);
-    }
-
-    /**
-     * Validates the provided input's primary site, histology and behavior. These properties are required to determine the cancer group and used at least in one of the rules in each group.
-     */
-    public static boolean validateProperties(String primarySite, String histology, String behavior) {
-        if (primarySite == null || primarySite.length() != 4 || !primarySite.startsWith("C") || !NumberUtils.isDigits(primarySite.substring(1)) || "C809".equalsIgnoreCase(primarySite)
-                || histology == null || histology.length() != 4 || !NumberUtils.isDigits(histology) || Integer.parseInt(histology) < 8000)
-            return false;
-        return !(behavior == null || !Arrays.asList("0", "1", "2", "3", "6").contains(behavior));
+    public List<MphGroup> getAllGroups() {
+        _groups.add(new Mp2007HeadAndNeckGroup());
```

Comment:
It's not safe to add the groups to a global variable like this whenever the method is called. You could end up with adding the groups twice in a multi-threaded environment. It would be better to add the groups in the constructor.